scheduler/src/cook/compute_cluster.clj

```diff
@@ -44,6 +44,12 @@
   (restore-offers [this pool-name offers]
     "Called when offers are not processed to ensure they're still available.")
 
+  (autoscaling? [this]
+    "Returns true if this compute cluster should autoscale to satisfy pending jobs")
```

Comment:
We should have just the one function `autoscale!` which is a no-op when the compute cluster does not support scaling.

Reply:
Currently using this function to filter down the list of compute clusters to those that can autoscale.